### PR TITLE
Fix LOGOUT_ON_PASSWORD_CHANGE settings not tested

### PR DESCRIPTION
**Pull request by SebCorbin**
- [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [x] All Python code must formatted using Black, and clean from pep8 and isort issues.
 - [x] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [x] If your changes are significant, please update `ChangeLog.rst`.
 - [x] If your change is substantial, feel free to add yourself to `AUTHORS`.

Let's test this and find out that it is bugged, then I'll ad a new commit to fix (TDD style)

### CODE REVIEW

File: allauth/account/tests/test_reset_password.py [hunk 1]
```diff
@@ -280,6 +280,17 @@ def test_password_reset_ACCOUNT_LOGIN_ON_PASSWORD_RESET(self):
         # EmailVerificationMethod.MANDATORY sends us to the confirm-email page
         self.assertRedirects(resp, "/confirm-email/")
 
+    @override_settings(ACCOUNT_LOGOUT_ON_PASSWORD_CHANGE=True)
+    def test_password_change_ACCOUNT_LOGOUT_ON_PASSWORD_CHANGE(self):
```
Comment (pennersr) [inline allauth/account/tests/test_reset_password.py:284]: We're in the process of moving to pytest, and therefore adding new old style tests is strongly discouraged. Can you turn this into a pytest? Just to get you started:

``` python
def test_passowrd_change_account_logout_on_password_change(settings, auth_client):
    settings.ACCOUNT_LOGOUT_ON_PASSWORD_CHANGE = True
    resp = auth_client.post(...)
    assert resp['location'] == ...

File: allauth/account/tests/test_reset_password.py [hunk 2]
```diff
@@ -300,3 +300,13 @@ def _create_user_and_login(self, usable_password=True):
...
+
+def test_password_change_ACCOUNT_LOGOUT_ON_PASSWORD_CHANGE(settings, auth_client, user_password):
+    settings.ACCOUNT_LOGOUT_ON_PASSWORD_CHANGE = True
+    data = {
+        "oldpassword": user_password, 
+        "password1": "newpass123", 
+        "password2": "newpass123"
+    }
+    resp = auth_client.post(reverse("account_change_password"), data)
+    assert resp['location'] == reverse("account_login")
```
Comment (pennersr) [inline allauth/account/tests/test_reset_password.py:312]: The build is failing on this point. Shouldn't you assert on `== settings.ACCOUNT_LOGOUT_REDIRECT_URL` ?

Comment (SebCorbin) [reply allauth/account/tests/test_reset_password.py]: Exactly yes, I mixed my project's settings for the defaults

File: allauth/account/views.py [hunk 3]
```diff
@@ -628,13 +628,7 @@ def dispatch(self, request, *args, **kwargs):
...
 
     def render_to_response(self, context, **response_kwargs):
-        if self.request.user.is_anonymous:
-            # We end up here when `ACCOUNT_LOGOUT_ON_PASSWORD_CHANGE = True`.
-            redirect_url = get_adapter(self.request).get_logout_redirect_url(
-                self.request
-            )
-            return HttpResponseRedirect(redirect_url)
-        elif not self.request.user.has_usable_password():
+        if not self.request.user.has_usable_password():
```
Comment (pennersr) [inline allauth/account/views.py:631]: Why did you alter this code? AFAIK, Django's `AnonymousUser` does not have a `has_usable_password()` method.

Comment (SebCorbin) [reply allauth/account/views.py:631]: AFAIK the comment _We end up here when `ACCOUNT_LOGOUT_ON_PASSWORD_CHANGE = True`_ is not valid (this is why I'm adding a test), which means : 
 - if we end up on this view as anonymous, we should be redirected somewhere, LOGIN_URL maybe ?
 - we're connected and if we submit this view and successfully change the password:
   - we have LOGOUT_ON_PASSWORD_CHANGE to True and should be logged out
   - LOGOUT_ON_PASSWORD_CHANGE is False and we just redirect to success_url

**Comment by pennersr** [comment]
Refactored via 088a4f1a

**Pull request closed without merging**